packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/base-types.ts

```diff
@@ -625,6 +625,196 @@ export interface TransformResources {
+   *
+   * @default - None
+   */
+  readonly containerHostName?: string;
+  /**
+   * The S3 path where the model artifacts, which result from model training, are stored.
+   * This path must point to a single gzip compressed tar archive (.tar.gz suffix).
+   * The S3 path is required for Amazon SageMaker built-in algorithms, but not if you use your own algorithms.
+   *
+   * @default - None
+   */
+  readonly modelS3Location?: S3Location;
+}
+
+/**
+ * Describes the container, as part of model definition.
+ *
+ * @see https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ContainerDefinition.html
+ * @experimental
+ */
+export class ContainerDefinition implements IContainerDefinition {
+
+  constructor(private readonly options: ContainerDefinitionOptions) {}
+
+  /**
+   * Called when the ContainerDefinition type configured on Sagemaker Task
+   */
+  public bind(task: ISageMakerTask): ContainerDefinitionConfig {
+    return {
+      parameters: {
+        ContainerHostname: this.options.containerHostName,
+        Image: this.options.image?.bind(task).imageUri,
+        Mode: this.options.mode,
+        ModelDataUrl: this.options.modelS3Location?.bind(task, { forReading: true }).uri,
+        ModelPackageName: this.options.modelPackageName,
+        Environment: this.options.environmentVariables?.value,
+      },
+    };
+  }
+}
+
+/**
+ * Configuration of the container used to host the model
+ *
+ * @see https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ContainerDefinition.html
+ * @experimental
+ */
+export interface IContainerDefinition {
+  /**
+   * Called when the ContainerDefinition is used by a SageMaker task.
+   */
+  bind(task: ISageMakerTask): ContainerDefinitionConfig;
+}
+
+/**
+ * Configuration options for the ContainerDefinition
+ */
+export interface ContainerDefinitionConfig {
+  /**
+   * Additional parameters to pass to the base task
+   *
+   * @default - No additional parameters passed
+   */
+  readonly parameters?: { [key: string]: any };
+}
+
+/**
+ * Specifies how many models the container hosts
+ *
+ * @experimental
+ */
+export enum Mode {
+  /**
+   * Container hosts a single model
+   */
+  SINGLE_MODEL = 'SingleModel',
+  /**
+   * Container hosts multiple models
+   *
+   * @see https://docs.aws.amazon.com/sagemaker/latest/dg/multi-model-endpoints.html
+   */
+  MULTI_MODEL = 'MultiModel',
+}
+
+/**
+ * Identifies a model that you want to host and the resources to deploy for hosting it.
+ *
+ * @see  https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ProductionVariant.html
+ * @experimental
+ */
+export interface ProductionVariant {
+  /**
+   * The size of the Elastic Inference (EI) instance to use for the production variant.
+   *
+   * @default - None
+   */
+  readonly acceleratorType?: AcceleratorType;
+  /**
+   * Number of instances to launch initially.
+   */
+  readonly initialInstanceCount: number;
```

Comment:
since the min value is one, would that be a sensible default here if we were to make it optional?